### ISSUE TITLE
remux video files but not audio files ?

### ISSUE THREAD

**Issue by chrizilla** [yt-dlp/yt-dlp#6655]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

What params achieve the following:
* remux all video containers to mkv
* don't remux audio files/containers

?

I currently use `--remux-video mkv` which achieves the first part (remux all videos to mkv) but how can I exempt all audio files ?

If possible, I'd like to avoid enumerating all audio formats, because it's not easy to have an exhaustive list.

Unfortunately there doesn't seem to be separate switches available for `--remxu-video` and `--remux-audio` which would be the best solution in order to be able to configure audio and video settings separately.

Thank you.

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by dirkf** [comment]
Which audio types (extensions) are you finding?

You might find that listing those before `mkv` (separate with `/`) causes them not to fall through to MKV.

**Comment by chrizilla** [comment]
> @dirkf 
> Which audio types (extensions) are you finding?

Any audio files really.

> You might find that listing those before `mkv` (separate with `/`) causes them not to fall through to MKV.

Thank you for your help. Is there no way to catch all audio files instead of listing all possible formats? I wanted to avoid that if possible:

> If possible, I'd like to avoid enumerating all audio formats, because it's not easy to have an exhaustive list.

**Comment by chrizilla** [comment]
@pukkandan: thank you for your help, too!

> Aren't you passing `-x` when downloading audio?

No. The helpfile says: _"`-x, --extract-audio` Convert video files to audio-only files"_
But I am talking about audio-only files (without video).

Examples:
I want to remux mp4, webm, etc -----> to mkv
I want to leave all audio-only files as is: mp3, m4a, etc.

My current settings convert all audio-only files (like mp3) to mkv.
So what I currently do is disable --remux-video in my config file before downloading an audio file and re-enable it afterwards.
But I suppose there is a more elegant way, no ?

> Just pass a no-op `--remux "mp4>mp4"` alongside it to override the previous remux options.
> You can even create an `--alias -x '--extract-audio --remux "mp4>mp4"'` if desired.

That sounds all very complicated. Is there no easy way to tell yt-dlp not to remux audio-onyl files ?
If not, I would like to file a feature request for that.
Maybe with the --remux-audio parameter suggested in the OP to be able to control the remuxing behaviour for audio and video files separately.
What do you think about that ?

**Comment by pukkandan** [comment]
> > Aren't you passing `-x` when downloading audio?
> 
> No. The helpfile says: _"`-x, --extract-audio` Convert video files to audio-only files"_ But I am talking about audio-only files (without video).
> 
> Examples: I want to remux mp4, webm, etc -----> to mkv I want to leave all audio-only files as is: mp3, m4a, etc.

It's not quite clear to me. Can you post your commands with examples?

**Comment by chrizilla** [comment]
@pukkandan : I'll try to explain again, but it is really easy, I don't know why I wasn't able to make myself understood. Could you please try to make sense of it? (Or if someone else reads this and would be so kind as to translate, I'd be of course equally thankful.)

My config file uses `--merge-output-format mkv` and  `--remux-video mkv` so all my downloads are remuxed to mkv, my preferred video container. Everything is fine so far.

My problem is that sometimes I download audio-only files (i.e. they have an audio track but no video track) which also get remuxed to mkv. **This** is the part I want to avoid.

So to summarize:
* I want to remux all video downloads to mkv.
* I want to keep the original container format for all audio-only downloads (i.e. not remux them to mkv). Examples: **mp3** files, **m4a** files, **opus** files, **wav** files, **flac** files etc.

Which settings do I have to use ?
As said in the OP: If possible, I'd like to avoid enumerating all audio formats one by one but rather tell yt-dlp to keep **ALL** audio containers.

Is it more understandable now ?
Thank you for looking into it.

**Comment by pukkandan** [comment]
That much is clear to me. And the answer is that it is not automatically possible currently. But my question is:

> My problem is that sometimes I download audio-only files

You must certainly be changing options to achieve this, no? When you do, couldn't you just add my above recommendation to it? I am trying to understand if it is worth converting this to a feature request. Examples with verbose log would certainly help me make a judgement. Idk why people are so hesitant to give them...

**Comment by chrizilla** [comment]
> @pukkandan : _the answer is that it is not automatically possible currently._ 

I am surprised that while yt-dlp is so highly configurable, such a basic task is not currently possible.

Ok, can you please make this issue a feature request then as per OP:

> Unfortunately there doesn't seem to be separate switches available for `--remxu-video` and `--remux-audio` which would be the best solution in order to be able to configure audio and video settings separately.

**Comment by chrizilla** [comment]
> You must certainly be changing options to achieve this, no?

Yes, I comment out `--merge-output-format mkv` and `--remux-video mkv` .

> When you do, couldn't you just add my above recommendation to it?

I don't understand how your parameters apply to my case. The helpfile says `"-x, --extract-audio Convert video files to audio-only files" ` but I am not trying to "extract" audio or "convert video files to audio-only files", so ...

> I am trying to understand if it is worth converting this to a feature request.

I am at a loss why it wouldn't be worth it? You have said yourself that currently it is not possible to automatically remux video to mkv (arguably the most popular container format out there) without also automatically remuxing ALL audio-only downloads to mkv ... which doesn't really make sense. I believe that most users downloading mp3, wma, or other audio files would prefer to keep them in their original container than to have them remuxed to mka. Or if it is not "most users" than at the very least not all users want to have all their audio downloads remuxed to mka. So please give us a way to avoid that.

> Examples with verbose log would certainly help me make a judgement.

I'll be more than happy to provide you with logs, but I have no idea what use case you would like a log of. Could you please give me a specific example for which I could provide a log?

**Comment by pukkandan** [comment]
> > You must certainly be changing options to achieve this, no?
> 
> Yes, I comment out `--merge-output-format mkv` and `--remux-video mkv` .

Then it will still download video, no?



> > Examples with verbose log would certainly help me make a judgement.
> 
> I'll be more than happy to provide you with logs, but I have no idea what use case you would like a log of. Could you please give me a specific example for which I could provide a log?

Of the above

**Comment by chrizilla** [comment]
> Then it will still download video, no?

The term "audio-only" appears 13 times in this issue.
And the title refers to audio files. Audio files don't have video.

> Of the above

Of what above? Please be specific as I really don't know what you want me to do.
Should I provide a log of any random audio download ?

**Comment by chrizilla** [comment]
Ok, it's a shot in the dark, because I don't know exactly what you are looking for, but here is a log of a audio download from a radio broadcast. The source is an mp3 stream but it unfortunately gets remuxed to mkv (which I want to avoid for audio-only files as desribed above).

```shell
[debug] Command-line config: ['https://fm4.orf.at/player/20230405/4MG']
[debug] Portable config "c:\hp\-\!portableapps\yt-dlp\yt-dlp.conf": ['--verbose', '--format-sort', 'hasvid,ie_pref,quality,res,hdr,codec,size,br,asr,ext,fps,hasaud,lang,source,proto,id', '-f', 'bestvideo*+bestaudio/best', '--paths', 'c:\\hp\\-\\#YT', '-o', '.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\%(title)s [%(extractor)s.%(extractor_key)s.%(id)s] [%(format_id)s].%(ext)s', '-o', 'thumbnail:.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\thumbs\\thumb.%(ext)s', '-o', 'subtitle:.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\subs\\%(alt_title,title)s [%(extractor)s.%(extractor_key)s.%(id)s] [%(format_id)s].%(ext)s', '-o', 'infojson:.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\!nfo\\%(alt_title,title)s [%(extractor)s.%(extractor_key)s.%(id)s] [%(format_id)s].%(ext)s', '-o', 'link:.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\!nfo\\%(title)s [%(extractor)s.%(extractor_key)s.%(id)s] [%(format_id)s].%(ext)s', '-o', 'description:.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\!nfo\\%(title)s [%(extractor)s.%(extractor_key)s.%(id)s] [%(format_id)s].%(ext)s', '-o', 'annotation:.\\%(playlist|)s\\%(playlist_index|)02d%(playlist_index& = |)s%(release_date>%Y-%m-%d,release_timestamp>%Y-%m-%d,upload_date>%Y-%m-%d,timestamp>%Y-%m-%d|#nodate#)s %(alt_title,title)s\\!nfo\\%(title)s [%(extractor)s.%(extractor_key)s.%(id)s] [%(format_id)s].%(ext)s', '-o', 'pl_infojson:.\\%(playlist|)s\\playlist.%(ext)s', '-o', 'pl_thumbnail:.\\%(playlist|)s\\folder.%(ext)s', '-o', 'pl_description:.\\%(playlist|)s\\playlist description.%(ext)s', '--output-na-placeholder', '~~~', '--write-info-json', '--write-description', '--write-annotations', '--no-overwrites', '--no-post-overwrites', '--merge-output-format', 'mkv', '--remux-video', 'mkv', '--write-subs', '--write-auto-subs', '--all-subs', '--sub-langs', 'all', '--sub-format', 'all/ttml/vtt/best', '--write-thumbnail', '--write-all-thumbnails', '--embed-metadata', '--embed-chapters', '--no-embed-info-json', '--write-link', '--write-url-link', '--write-desktop-link', '--no-check-certificate', '--retries', 'infinite', '--geo-bypass', '--geo-bypass-country', 'AT', '--no-clean-infojson', '--write-comments', '--audio-multistreams', '--no-playlist']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-105208-gb24f0c82b3-20220108 (setts), ffprobe N-105208-gb24f0c82b3-20220108
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Using fake IP 77.119.102.218 (AT) as X-Forwarded-For
[orf:radio] Extracting URL: https://fm4.orf.at/player/20230405/4MG
[orf:radio] 4MG: Downloading JSON metadata
[download] Downloading playlist: Morgengrauen
[info] Writing playlist metadata as JSON to: c:\hp\-\#YT\Morgengrauen\playlist.info.json
[info] Writing playlist description to: c:\hp\-\#YT\Morgengrauen\playlist description.description
[info] There's no playlist thumbnails to download
[orf:radio] Playlist Morgengrauen: Downloading 1 items of 1
[debug] The information of all playlist entries will be held in memory
[download] Downloading item 1 of 1
[debug] Sort order given by user: hasvid, ie_pref, quality, res, hdr, codec, size, br, asr, ext, fps, hasaud, lang, source, proto, id
[debug] Formats sorted by: hasvid, ie_pref, quality, res, hdr, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, fps, hasaud, lang, source, proto, id, channels
[info] 2023-04-05_0500_tl_54_7DaysWed1_139438: Downloading 1 format(s): 0
[info] Writing video description to: c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\!nfo\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].description
[info] There's no subtitles for the requested languages
[info] There's no video thumbnails to download
[info] Writing video metadata as JSON to: c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\!nfo\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].info.json
WARNING: There are no annotations to write.
[info] Writing internet shortcut (.url) to: c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\!nfo\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].url
[info] Writing internet shortcut (.desktop) to: c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\!nfo\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].desktop
[debug] Invoking http downloader on "https://loopstream01.apa.at/?channel=fm4&id=2023-04-05_0500_tl_54_7DaysWed1_139438.mp3"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mp3
[download] 100% of   82.49MiB in 00:06:11 at 227.28KiB/s
[VideoRemuxer] Remuxing video from mp3 to mkv; Destination: c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mkv
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mp3" -map 0 -dn -ignore_unknown -c copy -movflags "+faststart" "file:c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mkv"
Deleting original file c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mp3 (pass -k to keep)
[Metadata] Adding metadata to "c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mkv"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].mkv" -map 0 -dn -ignore_unknown -c copy -write_id3v1 1 -metadata "title=Morgengrauen" -metadata "date=20230405" -metadata "description=See the Sunrise: Aufstehen leicht gemacht mit dem FM4 Morgengrauen." -metadata "synopsis=See the Sunrise: Aufstehen leicht gemacht mit dem FM4 Morgengrauen." -metadata "purl=https://fm4.orf.at/player/20230405/4MG" -metadata "comment=https://fm4.orf.at/player/20230405/4MG" -movflags "+faststart" "file:c:\hp\-\#YT\Morgengrauen\01 = 2023-04-05 Morgengrauen\Morgengrauen [orf：radio.ORFRadio.2023-04-05_0500_tl_54_7DaysWed1_139438] [0].temp.mkv"
[info] Writing updated playlist metadata as JSON to: c:\hp\-\#YT\Morgengrauen\playlist.info.json
[download] Finished downloading playlist: Morgengrauen
```

**Comment by pukkandan** [comment]
Gotcha. I was thinking of downloading "audio-only" from youtube!

**Comment by chrizilla** [comment]
I see.  Ok, so what could be done about that?
Would you be open to the idea above ?
namely:
> the `--remux-audio` parameter suggested in the OP
> to be able to control the remuxing behaviour for audio and video files separately.

**Comment by chrizilla** [comment]
> Yes, I comment out `--merge-output-format mkv` and `--remux-video mkv` .

As a side-note: If I want to remux all video downloads to mkv, can/should I use both switches or is this overkill (possibly resulting in two remuxing operations) ? Does 1 switch suffice and if yes which one?

**Comment by dirkf** [comment]
Are there really are so many relevant audio types that a sensible config default couldn't be made with the existing options? It certainly wouldn't be longer than the output template in the log above.

**Comment by chrizilla** [comment]
Yes. From OP:
> avoid enumerating all audio formats, because it's not easy to have an exhaustive list

**Comment by dirkf** [comment]
From `utils.py` ll.6094-5, these `MEDIA_EXTENSIONS` could be considered
```py
    common_audio=('aiff', 'alac', 'flac', 'm4a', 'mka', 'mp3', 'ogg', 'opus', 'wav'),
    audio=('aac', 'ape', 'asf', 'f4a', 'f4b', 'm4b', 'm4p', 'm4r', 'oga', 'ogx', 'spx', 'vorbis', 'wma', 'weba'),
```
So this could be a stop-gap or actually an adequate solution:
```
--remux-video 'aiff/alac/flac/m4a/mka/mp3/ogg/opus/wav/aac/ape/asf/f4a/f4b/m4b/m4p/m4r/oga/ogx/spx/vorbis/wma/weba/mkv'
```
Or can only audio extensions "supported" by `--remux-video` be used here?

**Comment by chrizilla** [comment]
I am thankful for the help but I really beg you to consider a _real_ solution. There are some relevant audio formats missing in the list above and enumerating all possible audio formats is only a workaround. I am sure most users don't want to remux audio-only downloads.

**Comment by pukkandan** [comment]
> which would then require a `--remux-audio` parameter

No, that's what `--extract-audio` does. But

> `--remux-video` should not apply to audio-only files

this existing behavior cannot be changed for backward compat.

---

I understand your use case, but am not sure how to implement it. I am against adding more options if it can be helped. One idea could be to support syntax like `--remux audio>m4a,video>mp4`. But I need to think about it more - hence the label https://github.com/yt-dlp/yt-dlp/labels/triage

**Comment by chrizilla** [comment]
> > which would then require a `--remux-audio` parameter
> 
> No, that's what `--extract-audio` does.

I don't understand this. AFAIK:
* **muxing** = placing a media stream INTO a container
* **extracting** = extracting a media stream FROM a container
* **remuxing** = extracting a media stream from a containter and placing it into another container

Based on this, `--extract-audio` should do the opposite of what a potential `--remux-audio` parameter would do, no ?

And besides, the helpfile says:

`-x, --extract-audio             Convert video files to audio-only files`

... so why do you mention `--extract-audio` which _"converts video files to audio-only files"_, but we are already talking about audio-only files in the first place ?

> > `--remux-video` should not apply to audio-only files
> 
> this existing behavior cannot be changed for backward compat.

I generally agree, but are there no exceptions for bugs or unintended behaviour (which I think this is) ?



> I understand your use case, but am not sure how to implement it. I am against adding more options if it can be helped.

I generally agree. There are already so much options that it's hard to keep track.
But on the other hand this is such a basic case that it should be handled somehow.


> One idea could be to support syntax like `--remux audio>m4a,video>mp4`.

Looks good to me. Isn't that close what I suggested ?
How would the same command look like for `--remux audio>don't,video>mkv` ?

**Comment by pukkandan** [comment]
> > No, that's what `--extract-audio` does.
> 
> I don't understand this. AFAIK:
> ...
> And besides, the helpfile says:
> 
> `-x, --extract-audio Convert video files to audio-only files`
> 
> ... so why do you mention `--extract-audio` which _"converts video files to audio-only files"_, but we are already talking about audio-only files in the first place ?

What it actually does is convert the downloaded file into the given `--audio-format`[^1]. It performs this conversation whether the file is audio or video, making it a smart audio-only `--remux` + `--recode`

Throughout the documentation, "video" refers to audio and video collectively. I assume that is why the documentation was originally written like that. But I can see how it is misleading. Any suggestions for what to change it to?

[^1]: Or to a common_audio format that supports the downloaded codec if no `--audio-format` is given

**Comment by pukkandan** [comment]
> > One idea could be to support syntax like `--remux audio>m4a,video>mp4`.
> 
> Looks good to me. Isn't that close to what I suggested ?

Not really. It doesn't add new switches or change existing behaviors. We are instead reusing the existing `--remux` and `ext>ext` features.

> How would the same command look like for `--remux audio>donttouchme!,video>mkv` ?

Just `--remux video>mkv`

**Comment by chrizilla** [comment]
> > No, that's what `--extract-audio` does.
> 
> What it actually does is convert the downloaded file into the given `--audio-format`[1] It performs this conversation whether the file is audio or video, making it a smart audio-only `--remux` + `--recode`

Thanks for explaining. Unfortunately the terminology is a bit misleading then, because normally extraction refers to a **1:1 lossless** extraction, see for example the [mkvextract tool](https://mkvtoolnix.download/doc/mkvextract.html).


> Throughout the documentation, "video" refers to audio and video collectively. I assume that is why the documentation was originally written like that. But I can see how it is misleading. Any suggestions for what to change it to?

No this is not misleading. It's clear that a video file can have audio.

There seems to be another misunderstanding though:
We are talking about AUDIO-ONLY files here (**without video**).
So it doesn't make any sense to _"convert video files to audio-only files"_ with `--extract-audio` since we **already have** AUDIO-ONLY files WITHOUT VIDEO in the first place.


> It doesn't add new switches

Didn't you say above this is preferable ?
(_"I am against adding more options if it can be helped."_)

 
> > How would the same command look like for `--remux audio>donttouchme!,video>mkv` ?
> 
> Just `--remux video>mkv`

Looks good to me. So this idea is to keep `--remux-video` for compatibility and adding `--remux` as a new parameter ?

**Comment by pukkandan** [comment]
Oh, I see where u got confused! `--remux` is same option as `--remux-video`. Both work.

**Comment by chrizilla** [comment]
Ok, does this mean that I can use `--remux video>mkv` right now instead of `--remux-video mkv` (which you say is the same as `--remux mkv`) and the entire problem is solved ?

I don't think so, since above you said it's currently not possible.

**Comment by pukkandan** [comment]
No, I mean `--remux` and `--remux-video` are exactly the same option. i.e. `--remux mkv` = `--remux-video mkv`

**Comment by chrizilla** [comment]
Yes, I got that part and said as much: 

> `--remux-video mkv` (which you say is the same as `--remux mkv`)

**Comment by chrizilla** [comment]
My question was: 
> does this mean that I can use `--remux video>mkv` right now instead of `--remux-video mkv`
> and the entire problem is solved ?

Does `--remux video>mkv` mux all video files to mkv while leaving audio-only files untouched ?

**Comment by Erquint** [comment]
I find it quite dubious when the option `--remux-video` affects a file containing no video stream.

If we want to treat the term "video" in that option as a catchall for a stream container file and wish to avoid adding more options: just add `video` and `audio` as container class keywords for the `--remux-video` option, such that when the parser encounters those, it checks if the container carries (or expected to carry based on established format deliberations) a video stream.
There is no need to worry about backwards compatibility this way if users were never expected to use such container class keywords at any earlier point.

Example: `--remux-video "video>mkv/audio>ogg"`

**Comment by pukkandan** [comment]
@Erquint Yes, that was my suggestion as well. A PR implementing this would be welcome

> it checks if the container carries (or expected to carry based on established format deliberations) a video stream.

Audio-only formats (`audio` keyword) can be checked by `info['vcodec']='none'` and everything else falls under `video`


> I find it quite dubious when the option `--remux-video` affects a file containing no video stream.

We could deprecate (but not remove) the name in favor of simply `--remux`

**Comment by dontdieych** [comment]
```
yt-dlp \
  --extract-audio \
  --embed-thumbnail \
  --embed-metadata \
  --embed-chapters \
  --embed-subs \
  --embed-info-json \
  --no-overwrites \
  --remux-video mkv \
  --merge-output-format mkv \
  --convert-thumbnails png
```

This is my setup for downloading audio. Very nice but want to save as `.mka` extension. (or `m4a`)

Just little hope. :smile:
